### PR TITLE
187709040 v3 DI Item Search Requests

### PR DESCRIPTION
**Pull request by tealefristoe**
PT Story: https://www.pivotaltracker.com/story/show/187709040

This PR adds handlers for `delete` and `get itemSearch` API requests. The majority of the work took place in the `ResourceParser` and related utility functions, which parse the search query and find the appropriate items.

### CODE REVIEW

File: v3/src/data-interactive/resource-parser-utils.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { IDataSet } from "../models/data/data-set"
+import { DIParsedQuery, DIQueryFunction } from "./data-interactive-types"
+
+export function parseSearchQuery(query: string, dataContext?: IDataSet): DIParsedQuery {
+  if (query === "*") {
+    return { valid: true, func: () => true }
+  }
+
+  const matches = query.match(/([^=!<>]+)(==|!=|<=|<|>=|>)([^=!<>]+)/)
```
Comment (kswenson) [inline v3/src/data-interactive/resource-parser-utils.ts:9]: I assume this parsing code is taken from v2?

**Review by kswenson — APPROVED** [review]
👍 Looks good

File: v3/src/data-interactive/resource-parser-utils.ts (hunk 1 above)
Comment (tealefristoe) [reply v3/src/data-interactive/resource-parser-utils.ts:9]: Yes. Also on line 14, though there was a bug in that code in v2 that chatGPT helped me fix 👍  Should I add a comment mentioning where this code came from?

Comment (kswenson) [reply v3/src/data-interactive/resource-parser-utils.ts:9]: Seems like it could be useful.

**Pull request merged**